build.sh

```diff
@@ -0,0 +1,178 @@
+}
+
+_make_git_commit() {
+  pushd "${FOXHOUND_DIR}" > /dev/null || _die "Can't change into foxhound dir: ${FOXHOUND_DIR}"
+  NEW_BRANCHNAME="pw-build-$(date +%s)" 
+  git switch -c "${NEW_BRANCHNAME}" || _die "Can't create new Git branch called: ${NEW_BRANCHNAME}"
+  git add .
+  git commit --author="Foxhound builder <foxhound@whereismymind.info>" --message="Added Playwright support to Foxhound."
+  popd > /dev/null || exit 1
+}
+
+_prepare_playwright() {
+  pushd "${FOXHOUND_DIR}" > /dev/null || _die "Can't change into foxhound dir: ${FOXHOUND_DIR}"
+  if [ ! -f "${CURRENT_DIR}/.PLAYWRIGHT_VERSION" ]; then
+    _die "Can not determine playwright version, as \".PLAYWRIGHT_VERSION\" file is missing. Please open an issue on Github!"
+  fi
+  . "${CURRENT_DIR}/.PLAYWRIGHT_VERSION"
+  _status "Set playwright version to: ${PLAYWRIGHT_VERSION}"
+  popd > /dev/null || exit 1
+  pushd "${PLAYWRIGHT_DIR}" > /dev/null || _die "Can't change into playwright dir: ${PLAYWRIGHT_DIR}"
+  git fetch --all || _die "Git fetch failed for Playwright"
+  git checkout "${PLAYWRIGHT_VERSION}" || _die "Can't checkout playwright version ${PLAYWRIGHT_VERSION}.."
+  popd > /dev/null || exit 1
+}
+
+_checkout_playwright() {
+  _status "Cloning playwright into ${PLAYWRIGHT_DIR}"
+  git clone https://github.com/microsoft/playwright.git "${PLAYWRIGHT_DIR}" || _die "Cloning playwright failed!"
+}
+
+_check_repo_foxhound_repo_state() {
+  pushd "${FOXHOUND_DIR}" > /dev/null || _die "Can't change into foxhound dir: ${FOXHOUND_DIR}"
+ test -z "$(git status --porcelain)" || _die "Dirty Worktree, please commit all changes before running the build script!"
+  popd > /dev/null || exit 1
+}
+
+_prepare_foxhound() {
+  pushd "${FOXHOUND_DIR}" > /dev/null || _die "Can't change into foxhound dir: ${FOXHOUND_DIR}"
+ _status "Preparing the foxhound build environment"
+  if [ -d "./juggler" ]; then
+    _status "Deleting stale juggler"
+    rm -rf "${FOXHOUND_DIR}/juggler"
+  fi
+  if [ ! -f "${FOXHOUND_DIR}/.mozconfig" ]; then
+    _status "Setting default mozconfig from Ubuntu profile"
+    cp "${FOXHOUND_DIR}/taintfox_mozconfig_ubuntu" "${FOXHOUND_DIR}/.mozconfig" 
+  fi
+  if [ -n "$NO_CLOBBER" ]; then
+    ./mach --no-interactive clobber
+  fi
+ ./mach --no-interactive bootstrap --no-system-changes --application-choice=browser
+  popd > /dev/null || exit 1
+}
+
+_patch_foxhound() {
+  _status "Patching foxhound for playwright integration"
+  pushd "${FOXHOUND_DIR}" > /dev/null || _die "Can't change into foxhound dir: ${FOXHOUND_DIR}"
+  cp -r "${PLAYWRIGHT_DIR}/browser_patches/firefox/juggler" "juggler"
+  git apply --verbose --index --whitespace=nowarn --recount "${PLAYWRIGHT_DIR}/browser_patches/firefox/patches"/* || _die "Playwright patches failed to apply."
+  popd > /dev/null || exit 1
+}
+
+_build_foxhound() {
+  pushd "${FOXHOUND_DIR}" > /dev/null || _die "Can't change into foxhound dir: ${FOXHOUND_DIR}"
+  ./mach build || _die "./mach build error"
```

Comment:
Do you do any of the bootstrapping? Or is that left to be done by hand?

Reply:
OK, thanks, I must have missed it.

How do you handle the case where the bootstrap fails? We get this sometimes if the Mozilla build tools we need are no longer supported / hosted.

Reply:
hmmmm, I can add a flag only to skip the `./mach bootstrap` invocation. That seems like a sensible idea, currently, there is no error handling for bootstrap, but that is suboptimal too :)

Will fix this ASAP